File: nginx/cgi-bin/removeng.pl
#!/usr/bin/perl
my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = localtime(time());
$year += 1900;
$mon += 1;
$curdate = sprintf("%04d%02d%02d", $year, $mon, $mday );

$dir = '__data/Result';
opendir( $dh, $dir);

while (my $file = readdir $dh) {
    if($file =~ m/ng/ig){
	if($file =~ m/$curdate/ig){
	    # print "CUR : $file\n";	    
	}else{
	    my $unlink = "__data/Result/$file";
	    print "$unlink ";
	    if(-f $unlink){
		unlink($unlink);
		print " -- deleted \n";
	    }else{
		print " -- file error \n";
	    }
	}
    }else{
	# print "OK -- $file\n";
    }
}

closedir($dh);

File: nginx/cgi-bin/removeng.py
import os
import time

# Get the current date
sec, min, hour, mday, mon, year, wday, yday, isdst = time.localtime(time.time())
year += 1900
mon += 1
curdate = f"{year:04d}{mon:02d}{mday:02d}"

# Directory to search for files
dir = '__data/Result'
dh = os.listdir(dir)

for file in dh:
    # Check if the file name contains 'ng'
    if 'ng' in file.lower():
        # Check if the file name contains the current date
        if curdate in file:
            # print(f"CUR : {file}")
            pass
        else:
            unlink = f"__data/Result/{file}"
            print(unlink, end=' ')
            if os.path.isfile(unlink):
                os.remove(unlink)
                print(" -- deleted")
            else:
                print(" -- file error")
    else:
        # print(f"OK -- {file}")
        pass
